src/main/kotlin/org/rust/lang/core/lexer/RustLexerUtil.kt

```diff
@@ -1,17 +1,28 @@
 package org.rust.lang.core.lexer
 
+import com.intellij.lexer.Lexer
+
+fun CharSequence.tokenize(lexer: Lexer): Sequence<Lexer> =
+    generateSequence({
+        lexer.start(this)
+        lexer.tokenType?.let { lexer }
+    }, { currentLexer ->
+        currentLexer.advance()
+        currentLexer.tokenType?.let { currentLexer }
+    })
```

Comment:
Another option would be to yield pairs of `(tokenType, tokenText)`. `lexer.tokenEnd - lexer.tokenStart` is `tokenText.length`, right?